File: Top75/Array/53_ Maximum_Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSub = nums[0]
        currentSum = 0
        
        for n in nums:
            if currentSum < 0:
                currentSum = 0
                # What if there are only nagtive nums in this arr!?
                # That's why I think this is better...
                # https://www.youtube.com/watch?v=7J5rs56JBs8
            currentSum += n
            maxSub = max(currentSum,maxSub)
        return maxSub

File: Top75_JS/53_MaximumSubarray.js
function maxSubArray(nums) {
  let maxValue = nums[0];
  let accNum = nums[0];
  for (let i = 1; i < nums.length; i++) {
    let calc = Math.max(nums[i], accNum + nums[i]);
    if (calc > maxValue) maxValue = calc;
    accNum = calc;
  }
  return maxValue;
}
